build_centos.dockerfile

```diff
@@ -1,46 +1,42 @@
-# https://github.com/ghdl/ghdl/issues/1484
-# https://github.com/ghdl/ghdl/issues/1118
+# [build] Centos
 
-#
-# Centos 7
-#
+ARG IMAGE="centos:7"
 
-FROM centos:centos7
+#---
 
-RUN yum update -y && yum install -y git curl bzip2 wget flex fontconfig libX11 zlib-devel
-RUN yum install -y centos-release-scl && yum install -y devtoolset-8 texinfo
-SHELL [ "/usr/bin/scl", "enable", "devtoolset-8" ]
-
-#
-# Centos 8
-#
-
-#FROM centos:centos8
+FROM $IMAGE AS common
 
-#RUN yum update -y && yum install -y  git curl bzip2 wget flex fontconfig libX11 zlib-devel
-#RUN yum install -y --enablerepo=powertools gcc texinfo make
+RUN yum update -y \
+ && yum install -y \
+    bzip2 \
+    curl \
+    flex \
+    fontconfig \
+    git \
+    libX11 \
+    wget \
+    zlib-devel
 
-#
-# Common
-#
+#---
 
-WORKDIR /root
+FROM common AS gcc
 
-RUN mkdir gnat && \
-    curl -L https://community.download.adacore.com/v1/9682e2e1f2f232ce03fe21d77b14c37a0de5649b?filename=gnat-gpl-2017-x86_64-linux-bin.tar.gz | tar -xz -C gnat --strip-components=1
-RUN cd gnat && make ins-all prefix="build"
+RUN yum install -y centos-release-scl \
+ && yum install -y \
+    devtoolset-8 \
+    texinfo
 
-ENV PATH=/root/gnat/build/bin:$PATH
-
-RUN mkdir gcc && curl -L https://codeload.github.com/gcc-mirror/gcc/tar.gz/releases/gcc-8.3.0 | tar -xz -C gcc --strip-components=1
-RUN cd gcc && sed -i.bak s/ftp:/http:/g contrib/download_prerequisites && ./contrib/download_prerequisites && cd ..
+SHELL [ "/usr/bin/scl", "enable", "devtoolset-8" ]
 
-RUN git clone https://github.com/ghdl/ghdl.git
-RUN mkdir -p build/gcc-objs && cd build && ../ghdl/configure --with-gcc=../gcc --prefix=/usr/local && make copy-sources
-RUN cd build/gcc-objs && ../../gcc/configure --prefix=/usr/local --enable-languages=c,vhdl --disable-bootstrap --disable-lto --disable-multilib --disable-libssp --disable-libgomp --disable-libquadmath
-RUN cd build/gcc-objs && make -j2 && make install && cd .. && make ghdllib && make install
+# centos:8
+#RUN yum install -y --enablerepo=powertools \
+#    gcc \
+#    make \
+#    texinfo
 
-RUN ghdl --version
+RUN mkdir -p /tmp/gnat \
```

Comment:
We don't need to do this after installing version-specific packages, isn't it? That is, we can put this RUN statement in the firs stage (common).

Reply:
Done. I needed to install `make` in the first stage.